contracts/peripheral/ImportProxy.sol

```diff
@@ -0,0 +1,259 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.6.10;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../interfaces/IController.sol";
+import "../interfaces/IWeth.sol";
+import "../interfaces/IGemJoin.sol";
+import "../interfaces/IDaiJoin.sol";
+import "../interfaces/IVat.sol";
+import "../interfaces/IPool.sol";
+import "../interfaces/IFYDai.sol";
+import "../interfaces/IFlashMinter.sol";
+import "../helpers/DecimalMath.sol";
+import "../helpers/SafeCast.sol";
+import "../helpers/YieldAuth.sol";
+import "@nomiclabs/buidler/console.sol";
+
+interface IImportProxy {
+    function importFromProxy(IPool, address, uint256, uint256) external;
+    function hope(address) external;
+    function nope(address) external;
+}
+
+interface IProxyRegistry {
+    function proxies(address) external view returns (address);
+}
+
+contract ImportProxy is DecimalMath, IFlashMinter {
+    using SafeCast for uint256;
+    using YieldAuth for IController;
+
+    IVat public immutable vat;
+    IWeth public immutable weth;
+    IERC20 public immutable dai;
+    IGemJoin public immutable wethJoin;
+    IDaiJoin public immutable daiJoin;
+    IController public immutable controller;
+    address public immutable treasury;
+    IImportProxy public immutable splitter;
+    IProxyRegistry public immutable proxyRegistry;
+
+    bytes32 public constant WETH = "ETH-A";
+    bool public constant MTY = true;
+
+    constructor(IController controller_, IPool[] memory pools_, IProxyRegistry proxyRegistry_) public {
+        ITreasury _treasury = controller_.treasury();
+
+        weth = _treasury.weth();
+        dai = _treasury.dai();
+        daiJoin = _treasury.daiJoin();
+        wethJoin = _treasury.wethJoin();
+        vat = _treasury.vat();
+
+        controller = controller_;
+        treasury = address(_treasury);
+        splitter = IImportProxy(address(this)); // This contract has two functions, as itself, and delegatecalled by a dsproxy.
+
+        proxyRegistry = proxyRegistry_;
+
+        // Allow pool to take fyDai for trading
+        for (uint i = 0 ; i < pools_.length; i++) {
+            pools_[i].fyDai().approve(address(pools_[i]), type(uint256).max);
+        }
+
+        // Allow treasury to take weth for posting
+        _treasury.weth().approve(address(_treasury), type(uint256).max);
```

Comment:
Isn't this cheaper?

```suggestion
        weth.approve(address(_treasury), type(uint256).max);
```

Reply:
No can do, because `weth` is immutable, so it can't be called from within the constructor. I fixed it with some local variables.